src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs

```diff
@@ -674,7 +674,34 @@ internal string GetString()
                     FlushBytes();
                 }
 
-                return _charBuffer.Slice(0, _numChars).ToString();
+                Span<char> chars = _charBuffer.Slice(0, _numChars);
+
+                const char HIGH_SURROGATE_START = '\ud800';
+                const char LOW_SURROGATE_END = '\udfff';
+
+                // Replace any invalid surrogate chars.
+                for (int i = chars.IndexOfAnyInRange(HIGH_SURROGATE_START, LOW_SURROGATE_END); (uint)i < (uint)chars.Length; i++)
```

Comment:
Slight personal preference for readability, feel free to ignore if we do not do it in other places.

```suggestion
                int idxOfFirstSurrogate = chars.IndexOfAnyInRange(HIGH_SURROGATE_START, LOW_SURROGATE_END);
                for (int i = idxOfFirstSurrogate; (uint)i < (uint)chars.Length; i++)
```